cmd/promtool/main.go

```diff
@@ -364,6 +384,42 @@ func (ls lintConfig) lintDuplicateRules() bool {
 	return ls.all || ls.duplicateRules
 }
 
+const promDefaultURL = "http://localhost:9090"
+
+// Check server status - healthy & ready.
+func CheckServerStatus(serverURL, checkEndpoint string) error {
+	if serverURL == "" {
+		serverURL = promDefaultURL
+	}
+
+	u, err := url.Parse(serverURL)
+	if err != nil {
+		return fmt.Errorf("error parsing URL: %s", serverURL)
+	}
+
+	u.Path += checkEndpoint
+
+	serverURL = u.String()
+
+	// Check Health & Readiness.
+	res, err := http.Get(serverURL)
+	if err != nil {
+		return err
+	}
```

Comment:
Some feedback:

1. This fails for self signed certificates:

```shell
~/go/src/github.com/prometheus/prometheus (issue-12000 [  ]) -> ./promtool-12000 check health "https://$PROM:9090"
Get "https://REDACTED:9090/-/healthy": tls: failed to verify certificate: x509: certificate signed by unknown authority
```

2. This is a bit cumbersome to use with basic auth, as you need to explicitly embed the `user:password@server` into the url. Prometheus now supports basic auth natively, so it would be nice to make it easier to provide credentials for users.

The good news is that both of these can be solved with [HTTP Config Files](https://prometheus.io/docs/alerting/latest/configuration/#http_config), and there's already framework for supporting HTTP config files elsewhere in promtool already! (See [`httpRoundTripper`](https://github.com/prometheus/prometheus/blob/58a8d526e8f1dbe830997900d91483c35fd218ea/cmd/promtool/main.go#L80) and [`httpConfigFilePath`](https://github.com/prometheus/prometheus/blob/58a8d526e8f1dbe830997900d91483c35fd218ea/cmd/promtool/main.go#L82)).

Reply:
I have added support for self signed certificates and here is the respective output:
```
./promtool check health http://192.168.149.221:9090
 SUCCESS: Prometheus is Healthy.

./promtool check health https://localhost:9090 --http.config.file=<path>/http-config.yml
  SUCCESS: Prometheus Server is Healthy. 

./promtool check ready http://192.168.149.221:9090
  SUCCESS: Prometheus is Ready.

./promtool check ready https://localhost:9090 --http.config.file=<path>/http-config.yml
  SUCCESS: Prometheus Server is Ready.
```